hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStateStore.java

```diff
@@ -352,8 +352,16 @@ public static State getRegionState(final Result r, int replicaId) {
     if (cell == null || cell.getValueLength() == 0) {
       return null;
     }
-    return State.valueOf(Bytes.toString(cell.getValueArray(), cell.getValueOffset(),
-        cell.getValueLength()));
+
+    String state = Bytes.toString(cell.getValueArray(), cell.getValueOffset(),
+        cell.getValueLength());
+    try {
+      return State.valueOf(state);
+    }
+    catch (IllegalArgumentException e) {
+      LOG.debug("BAD value {} in hbase:meta info:state column", state);
```

Comment:
Can we also log the specific region encoded name? Also, as it's an exceptional situation, I think log level should be WARN. We could also give some tip for fixing it using hbck2 _setRegionStateInMeta_ command?

Reply:
What @wchevreuil  said